### ISSUE TITLE
Add a Qt wrapper module

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#1076]
This might be useful when [PySide2](https://github.com/PySide/pyside2) is more usable - and we definitely should have it in place before starting with the plugin API.

We also could simplify `@pyqtSlot` by using type annotations, like in [this example](https://gist.github.com/empyrical/a103c46f454a433e8798) (we probably don't need all the overloading logic).

**Comment by The-Compiler** [comment]
Duplicate of #995.
